Dockerfile

```
@@ -443,29 +443,73 @@ RUN \
 
 FROM sdk-libc as sdk-go-prep
 
-ENV GOVER="1.21.9"
+# Set up the environment for building.
+ENV GOOS="linux"
+ENV CGO_ENABLED=1
+ENV CFLAGS="-O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-clash-protection"
+ENV CXXFLAGS="${CFLAGS}"
+ENV LDFLAGS="-Wl,-z,relro -Wl,-z,now"
+ENV CGO_CFLAGS="${CFLAGS}"
+ENV CGO_CXXFLAGS="${CXXFLAGS}"
+ENV CGO_LDFLAGS="${LDFLAGS}"
+
+ENV GO111MODULE="auto"
+
 ENV AWS_LC_FIPS_VER="2.0.9"
 
 USER root
 RUN dnf -y install golang
 
+ENV GOVER_1_21="1.21.9"
+ENV GOVER_1_22="1.22.1"
+
+# =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+FROM sdk-go-prep as sdk-go-1.21-prep
+
 USER builder
-WORKDIR /home/builder/sdk-go
-COPY ./hashes/go /home/builder/hashes
+
+WORKDIR /home/builder/sdk-go-${GOVER_1_21}
+
+COPY ./hashes/go-${GOVER_1_21} /home/builder/hashes
+COPY ./helpers/go/* ./
+COPY patches/go-${GOVER_1_21}/* ./
+
+RUN ./prep-go.sh --go-version=${GOVER_1_21}
+
+# We need to build AWS-LC before we can build Go.
+WORKDIR /home/builder/aws-lc
+COPY ./hashes/aws-lc /home/builder/hashes
 RUN \
   sdk-fetch /home/builder/hashes && \
-  tar --strip-components=1 -xf go${GOVER}.src.tar.gz && \
-  rm go${GOVER}.src.tar.gz
-
-# Patch Go sources so that they work with AWS-LC as the crypto implementation.
-# Note that this will break use of `GOEXPERIMENT=boringcrypto` when using the
-# default syso files that ship with Go, since the functions and data structures
-# will no longer match. We build the replacement AWS-LC syso files below.
-COPY patches/go/* ./
+  tar --strip-components=1 -xf AWS-LC-FIPS-${AWS_LC_FIPS_VER}.tar.gz && \
+  rm AWS-LC-FIPS-${AWS_LC_FIPS_VER}.tar.gz
+
+# Patch AWS-LC sources to avoid weak symbols for memory management functions
+# when GOBORING is defined.
+COPY patches/aws-lc/* ./
 RUN \
   git init && \
   git apply --whitespace=nowarn *.patch
 
+WORKDIR /home/builder/aws-lc/build
+COPY ./configs/aws-lc/* .
+COPY ./helpers/aws-lc/* .
+
+# =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+FROM sdk-go-prep as sdk-go-1.22-prep
+
+USER builder
+
+WORKDIR /home/builder/sdk-go-${GOVER_1_22}
+
+COPY ./hashes/go-${GOVER_1_22} /home/builder/hashes
```

```suggestion
COPY ./hashes/go-${GOVER_1_22} /home/builder/hashes-go`
```

By using `hashes-go` and `hashes-aws-lc` you can make both files available in `prep-go.sh`.